plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php

```diff
@@ -620,11 +635,19 @@ private function create_customer_account( $user_email, $first_name, $last_name )
 			throw new \Exception( 'registration-error-email-exists' );
 		}
 
-		$username = wc_create_new_customer_username( $user_email );
+		// Handle password creation if not provided.
+		if ( empty( $password ) ) {
+			$password           = wp_generate_password();
+			$password_generated = true;
+		} else {
+			$password_generated = false;
+		}
 
-		// Handle password creation.
-		$password           = wp_generate_password();
-		$password_generated = true;
+		if ( empty( $password ) ) {
```

Comment:
Will this ever be reachable since we have the same condition a few lines back generating a password?

Reply:
I am coding defensively because `wp_generate_password` is filterable.

Reply:
Got it, good call! Can we add a comment why so this wouldn't be lost in a future change?

Reply:
Will do!

Reply:
https://github.com/woocommerce/woocommerce/pull/48985/commits/a39134e3d1d864955791314c663e6f94850db429